src/controller/selectionpostfixer.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module engine/controller/selectionpostfixer
+ */
+
+import Range from '../model/range';
+import Position from '../model/position';
+
+/**
+ * The selection post fixer which check if nodes with `isLimit` property in schema are properly selected.
+ *
+ * See as an example selection that starts in P1 element and ends inside text of TD element
+ * (`[` and `]` are range boundaries and `(l)` denotes element defines as `isLimit=true`):
+ *
+ *		root
+ *		 |- element P1
+ *		 |   |- "foo"                                      root
+ *		 |- element TABLE (l)                   P1         TABLE             P2
+ *		 |   |- element TR (l)                 f o[o     TR      TR         b a r
+ *		 |   |   |- element TD (l)                       TD      TD
+ *		 |   |       |- "aaa"                          a]a a    b b b
+ *		 |   |- element TR (l)
+ *		 |   |   |- element TD (l)                           ||
+ *		 |   |       |- "bbb"                                ||
+ *		 |- element P2                                       VV
+ *		 |   |- "bar"
+ *		                                                   root
+ *		                                        P1         TABLE]            P2
+ *		                                       f o[o     TR      TR         b a r
+ *		                                                 TD      TD
+ *		                                               a a a    b b b
+ *
+ * In above example, the TABLE, TR and TD are defined as `isLimit=true` in the schema. The range that is not contained withing
+ * single limit element must be expanded to select outer most parent limit element. The range end is inside text node of TD element.
+ * As TD element is a child of TR element and TABLE elements which both are defined as `isLimit=true` in schema the range must be expanded
+ * to select whole TABLE element.
+ *
+ * **Note** If selection contains multiple ranges the method returns minimal set of ranges that are not intersecting after expanding them
+ * to select `isLimit=true` elements.
+ *
+ * See {@link module:engine/model/schema~Schema#isLimit}.
+ *
+ * @param {module:engine/model/writer~Writer} writer
+ * @param {module:engine/model/model~Model} model
+ */
+export default function selectionPostFixer( writer, model ) {
```

Comment:
It's a function so its name should start with a verb. You can find similar logic in the view – e.g. there's `view/filler~injectQuirksHandling`. 

I'd propose to move this code to `model/utils/selection-postfixer` and expose a function called `injectSelectionPostfixer`. I'd also say that it can be used in `Model#constructor()` because:

* it will be symmetrical to `View#constructor()` injecting some behaviours,
* it's related to schema and basic schema is defined in `Model`.

Reply:
K, thanks for pointing out proper placement for the logic. I'll update the PR with this change.

Reply:
Actually, I see that this function doesn't add a postfixer itself. Check out how `view/filler.js` is structured – this way of exposing filler's functionality encapsulates what it needs to do – how many listeners it has to register, with what params, etc.

In this case, we're talking about just one post fixer and registering it in a straightforward way via `doc.registerPostFixer()`. Therefore, on one hand, exporting the callback like you did seems sufficient. OTOH, it's hard to tell now whether this is all that we'll ever need. Hence... IDK :P I leave it up to you.